.trunk/trunk

```diff
@@ -0,0 +1,186 @@
+#!/bin/bash
+
+set -euo pipefail
+
+platform=$(uname | tr "[:upper:]" "[:lower:]")
+if [ "${platform}" == "darwin" ]; then
+  sha256sum() { shasum -a 256 "$@"; }
+fi
+
+TRUNK_LAUNCHER_VERSION=0.1.31-alpha
+
+check_mark="\033[0;32m✔\033[0m"
+x_mark="\033[0;31m✘\033[0m"
+
+download() {
+  url=${1}
+  sha256=${2}
+  downloaded=${3}
+  download_name=${4}
+
+  downloading_text="Downloading ${download_name}..."
+  echo "${downloading_text}"
+
+  download_success=false
+  # Download via wget or curl. wget does retries by default, but we configure them for curl.
+  if [ -x "$(which wget)" ]; then
+    if wget "${url}" -O "${downloaded}" --quiet --show-progress; then
+      echo -e "\033[1A\033[K\033[1A\r${check_mark} ${downloading_text} done"
+      download_success=true
+    fi
+  elif [ -x "$(which curl)" ]; then
+    if curl --connect-timeout 5 --max-time 10 --retry 3 --retry-delay 0 --retry-max-time 60 "${url}" --output "${downloaded}" --fail; then
+      echo -e "${check_mark} ${downloading_text} done"
+      download_success=true
+    fi
+  else
+    echo -e "\033[1A\r${x_mark} ${downloading_text} FAILED"
+    echo -e "${x_mark} Unable to download without curl or wget installed."
+    exit 1
+  fi
+
+  if ! $download_success; then
+    echo -e "${x_mark} Unable to download trunk. Please check your connection, try again, see https://status.trunk.io, or report this issue at https://slack.trunk.io"
+    exit 1
+  fi
+
+  verifying_text="Verifying ${download_name} sha256..."
+  echo -n "${verifying_text}"
+  actual_line=$(sha256sum "${downloaded}")
+  actual=${actual_line%% *}
+  if [[ ${actual} != "${sha256}" ]]; then
+    echo -e "\r${x_mark} ${verifying_text} FAILED"
+    echo "Expected sha256: '${sha256}'"
+    echo "Actual sha256:   '${actual}'"
+    exit 1
+  fi
+  echo -e "\r${check_mark} ${verifying_text} done"
+}
+
+add_to_bin() {
+  bin_script="/usr/local/bin/trunk"
+  bash_source=${BASH_SOURCE[0]}
+
+  if [[ -f ${bin_script} ]] && [[ ${bin_script} != "${bash_source}" ]]; then
+    read -rp "Trunk Launcher is already installed in /usr/local/bin. Would you like to override it? (Y/n) " yn
```

Comment:
Having a "launcher" and then a separate binary (?) is worrisome.

This script is interactive, which isn't appropriate for us.